src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -133,6 +135,28 @@ def _mark_as_time_ineligible(charges: List[Charge], reason: str, eligibility_dat
         for charge in charges:
             charge.set_time_ineligible(reason, eligibility_date)
 
+    @staticmethod
+    def _set_arrests_as_eligible(charges_with_summary: ChargesWithSummary):
+        # If most recent conviction is time eligible
+        # AND all other charges in the same case are arrests
+        # Then make all other charges in the case time eligible
+        mrc = charges_with_summary.most_recent_conviction
+        second_mrc = charges_with_summary.second_most_recent_conviction
+        if mrc and mrc.expungement_result.time_eligibility and mrc.expungement_result.time_eligibility.status is EligibilityStatus.ELIGIBLE:
+            if not second_mrc:
+                for charge in mrc.case()().charges:
+                    if charge.acquitted():
+                        charge.set_time_eligible()
+        else:
+            if charges_with_summary.second_most_recent_conviction and not mrc is None:
+                for charge in mrc.case()().charges:
+                    if second_mrc.disposition and charge.acquitted():
+                            charge.set_time_ineligible(
+                                'Multiple convictions within last ten years',
```

Comment:
Please reference `137.225 7(b)` somehow here.

Reply:
These reason strings are all inconsistent but they can be fixed in a separate PR addressing the issue: #619